### PR TITLE
feat: added custom keystore/truststore load in spring boot zeebe client

### PR DESCRIPTION
**Pull request by igpetrov**
## Description

feat: added custom keystore/truststore load in spring boot zeebe client

There are cases where a user doesn't want to maintain whole SSL infrastructure for the Camunda cluster but also use X509 authorizes.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

Related to: https://github.com/camunda/product-hub/issues/2451

### CODE REVIEW

File: clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/CredentialsProviderSelfManagedWithSSLTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  }
+
+  @Test
+  void shouldBeSelfManaged() {
+    final CredentialsProvider credentialsProvider = configuration.getCredentialsProvider();
+    assertThat(credentialsProvider).isExactlyInstanceOf(OAuthCredentialsProvider.class);
+  }
+
+  @Test
+  void shouldHaveZeebeAuth() throws IOException {
```
Comment (igpetrov) [inline clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/CredentialsProviderSelfManagedWithSSLTest.java:124]: Hi @ana-vinogradova-camunda,

I am adding a new test case for identity provider only SSL support. I see a regular test class has the [same test disabled](https://github.com/camunda/camunda/blob/966f5c1b97a839da57c87757fb435cdd03be1004/clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/CredentialsProviderSelfManagedTest.java#L94). I am wondering, why that test is disables and should I do the same? Should I even include it? Thank you!

**Comment by igpetrov** [comment]
Hi @nicpuppa 

This one is ready for review. Thank you!

**Review by nicpuppa — APPROVED** [review]
Looks good 👍 

❓ Are you taking care also of the documentation ? For both clients

**Comment by igpetrov** [comment]
> Looks good 👍
> 
> ❓ Are you taking care also of the documentation ? For both clients

Thank you for review. Yes, that's my next task as well

File: clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/CredentialsProviderSelfManagedWithSSLTest.java (hunk 1 above)
Comment (igpetrov) [reply clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/CredentialsProviderSelfManagedWithSSLTest.java:124]: Ah, I see. Never observed it failing locally for both instances. I'll take a look if I can help with that

**Pull request merged**